service/history/decisionChecker.go

```diff
@@ -452,3 +462,53 @@ func (v *decisionAttrValidator) validateStartChildExecutionAttributes(
 
 	return nil
 }
+
+func (v *decisionAttrValidator) validateCrossDomainCall(
```

Comment:
Are there reasons to expose new method ? Can you not automatically embed this validation within the existing validateFooAttributes() methods ? I like the fact that there is one validate method for a given attribute type. That way, caller code doesn't have to change when we add new types of validations for the same set of attributes.

Reply:
move this check to within corresponding decision check